source/daplink/cmsis-dap/DAP.c

```diff
@@ -1467,10 +1547,12 @@ uint32_t DAP_ProcessCommand(const uint8_t *request, uint8_t *response) {
       break;
 
     case ID_DAP_Connect:
+			if (!DAP_Lock(0)) return 0;
```

Comment:
If the disconnect message is missed then DAP will remain locked. There may need to be some mechanism to timeout or to steal ownership of the lock. A similar problem also applies CDC.